geoportailv3/static/js/search/searchdirective.js

```diff
@@ -0,0 +1,516 @@
+  };
+
+  /**
+   * @type {Array.<string>}
+   * @private
+   */
+  this.showGeom_ = [
+    'hydro',
+    'Adresse',
+    'FLIK',
+    'biotope',
+    'editus_poi_285',
+    'editus_poi_286',
+    'editus_poi_287',
+    'editus_poi_289',
+    'editus_poi_290',
+    'editus_poi_291',
+    'editus_poi_292',
+    'editus_poi_293',
+    'editus_poi_294',
+    'editus_poi_295',
+    'editus_poi_296',
+    'editus_poi_297',
+    'editus_poi_298',
+    'editus_poi_299',
+    'hydro_km',
+    'asta_esp',
+    'Parcelle'
+  ];
+  /**
+   * @type {Array.<ol.layer.Layer>}
+   * @private
+   */
+  this.layers_ = [];
+
+  /**
+   * @type {ngeo.BackgroundLayerMgr}
+   * @private
+   */
+  this.backgroundLayerMgr_ = ngeoBackgroundLayerMgr;
+
+  /**
+   * @type {ol.FeatureOverlay}
+   * @private
+   */
+  this.featureOverlay_ = this.createFeatureOverlay_();
+
+  /**
+   * @type {app.GetLayerForCatalogNode}
+   * @private
+   */
+  this.getLayerFunc_ = appGetLayerForCatalogNode;
+
+  /**
+   * @type {app.ShowLayerinfo}
+   * @private
+   */
+  this.showLayerinfo_ = appShowLayerinfo;
+
+  /** @type {Bloodhound} */
+  var POIBloodhoundEngine = this.createAndInitPOIBloodhound_(
+      ngeoCreateGeoJSONBloodhound, searchServiceUrl);
+
+  /** @type {Bloodhound} */
+  var LayerBloodhoundEngine = this.createAndInitLayerBloodhound_();
+
+  /** @type {Bloodhound} */
+  var BaseLayerBloodhoundEngine = this.createAndInitLayerBloodhound_();
+
+  $scope.$on('gettextLanguageChanged', goog.bind(function(event, args) {
+    this.createLocalAllLayerData_(
+        appThemes, LayerBloodhoundEngine, gettextCatalog);
+    this.createLocalBaseLayerData_(
+        appThemes, BaseLayerBloodhoundEngine, gettextCatalog);
+  }, this));
+
+
+  /** @type {TypeaheadOptions} */
+  this['options'] = {
+    highlight: true
+  };
+
+  /** @type {Array.<TypeaheadDataset>} */
+  this['datasets'] = [{
+    source: POIBloodhoundEngine.ttAdapter(),
+    displayKey: function(suggestion) {
+      var feature = /** @type {ol.Feature} */ (suggestion);
+      return feature.get('label');
+    },
+    templates: /** @type {TypeaheadTemplates} */({
+      header: function() {
+        return '<div class="header">' +
+            gettextCatalog.getString('Addresses') +
+            '</div>';
+      },
+      suggestion: function(suggestion) {
+        var feature = /** @type {ol.Feature} */ (suggestion);
+
+        // A scope for the ng-click on the suggestion's « i » button.
+        var scope = $scope.$new(true);
+        scope['feature'] = feature;
+        scope['click'] = function(event) {
+          event.stopPropagation();
+        };
+
+        var html = '<p>' + feature.get('label') +
+            ' <span>(' + gettextCatalog.getString(
+                /** @type {string} */ (feature.get('layer_name'))
+            ) + ')</span></p>';
+        return $compile(html)(scope);
+      }
+    })
+  },{
+    source: LayerBloodhoundEngine.ttAdapter(),
+    displayKey: function(suggestion) {
```

Comment:
In this case its not a BackgroundLayerSuggestion object though, but rather a catalog node (ie normal layer), which was not explicitely typed afaik